mlflow/store/artifact/presigned_url_artifact_repo.py

```diff
@@ -0,0 +1,165 @@
+import os
+import posixpath
+from collections import namedtuple
+from typing import List
+
+import mlflow
+from mlflow.protos.databricks_artifacts_pb2 import ArtifactCredentialInfo
+from mlflow.protos.databricks_fs_service_pb2 import FilesystemService, CreateDownloadUrlRequest, \
+    CreateDownloadUrlResponse, ListRequest, ListResponse
+from mlflow.protos.databricks_uc_registry_service_pb2 import UcModelRegistryService
+from mlflow.store.artifact.cloud_artifact_repo import CloudArtifactRepository, StagedArtifactUpload
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.file_utils import download_file_using_http_uri
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import call_endpoint, extract_api_info_for_service, _REST_API_PATH_PREFIX
+
+_METHOD_TO_INFO_UCMR = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
+_METHOD_TO_INFO_FS = extract_api_info_for_service(FilesystemService, _REST_API_PATH_PREFIX)
+
+PresignedUrlAndHeaders = namedtuple(
+    "PresignedUrlAndHeaders",
+    [
+        # presigned URL for the artifact
+        "url",
+        # headers to include in http request when using the presigned URL
+        "headers",
+    ],
+)
+
+
+class PresignedUrlArtifactRepository(CloudArtifactRepository):
+    """
+    Stores artifacts on S3 using presigned URLs.
+    """
+
+    def __init__(self, artifact_uri):
+        super().__init__(artifact_uri)
+
+    def log_artifact(self, local_file, artifact_path=None):
+        raise NotImplementedError("this is for testing purposes only, not for production use.")
```

Comment:
Let's have a better error message here.  At a minimum, it should just be `log_artifact